packages/audioplayers/tizen/src/audio_player.cc

```diff
@@ -3,111 +3,10 @@
-  std::string result;
-  switch (code) {
-    case PLAYER_ERROR_NONE:
-      result = "player - Successful";
-      break;
-    case PLAYER_ERROR_OUT_OF_MEMORY:
-      result = "player - Out of memory";
-      break;
-    case PLAYER_ERROR_INVALID_PARAMETER:
-      result = "player - Invalid parameter";
-      break;
-    case PLAYER_ERROR_NO_SUCH_FILE:
-      result = "player - No such file or directory";
-      break;
-    case PLAYER_ERROR_INVALID_OPERATION:
-      result = "player - Invalid operation";
-      break;
-    case PLAYER_ERROR_FILE_NO_SPACE_ON_DEVICE:
-      result = "player - No space left on the device";
-      break;
-    case PLAYER_ERROR_FEATURE_NOT_SUPPORTED_ON_DEVICE:
-      result = "player - Not supported Feature";
-      break;
-    case PLAYER_ERROR_SEEK_FAILED:
-      result = "player - Seek operation failure";
-      break;
-    case PLAYER_ERROR_INVALID_STATE:
-      result = "player - Invalid state";
-      break;
-    case PLAYER_ERROR_NOT_SUPPORTED_FILE:
-      result = "player - File format not supported";
-      break;
-    case PLAYER_ERROR_INVALID_URI:
-      result = "player - Invalid URI";
-      break;
-    case PLAYER_ERROR_SOUND_POLICY:
-      result = "player - Sound policy error";
-      break;
-    case PLAYER_ERROR_CONNECTION_FAILED:
-      result = "player - Streaming connection failed";
-      break;
-    case PLAYER_ERROR_VIDEO_CAPTURE_FAILED:
-      result = "player - Video capture failed";
-      break;
-    case PLAYER_ERROR_DRM_EXPIRED:
-      result = "player - Expired license";
-      break;
-    case PLAYER_ERROR_DRM_NO_LICENSE:
-      result = "player - No license";
-      break;
-    case PLAYER_ERROR_DRM_FUTURE_USE:
-      result = "player - License for future use";
-      break;
-    case PLAYER_ERROR_DRM_NOT_PERMITTED:
-      result = "player - Format not permitted";
-      break;
-    case PLAYER_ERROR_RESOURCE_LIMIT:
-      result = "player - Resource limit";
-      break;
-    case PLAYER_ERROR_PERMISSION_DENIED:
-      result = "player - Permission denied";
-      break;
-    case PLAYER_ERROR_SERVICE_DISCONNECTED:
-      result = "player - Socket connection lost";
-      break;
-    case PLAYER_ERROR_BUFFER_SPACE:
-      result = "player - No buffer space available";
-      break;
-    case PLAYER_ERROR_NOT_SUPPORTED_AUDIO_CODEC:
-      result = "player - Not supported audio codec but video can be played";
-      break;
-    case PLAYER_ERROR_NOT_SUPPORTED_VIDEO_CODEC:
-      result = "player - Not supported video codec but audio can be played";
-      break;
-    case PLAYER_ERROR_NOT_SUPPORTED_SUBTITLE:
-      result = "player - Not supported subtitle format";
-      break;
-    default:
-      result = "player - Unkonwn error";
-      break;
-  }
-  return result;
+static std::string ErrorToString(int code) {
+  const char *ptr = get_error_message(code);
+  std::string result = ptr ? std::string(ptr) : "";
+  return "player - " + (result.empty() ? "Unknown error" : result);
```

Comment:
Wouldn't it be better to remove this function itself and just print errors like:

```cpp
LOG_ERROR("error blabla: %s", get_error_message(code));
```

Reply:
If we didn't need a string type, then yes, but the returned string type is also used as a parameter in these places in `audio_player.cc`:

line 131 ~136

```c++
if (result != PLAYER_ERROR_NONE) {
  seeking_ = false;
  std::string error = ErrorToString(result);
  LOG_ERROR("player_set_play_position failed : %s", error.c_str());
  throw AudioPlayerError(error, "player_set_play_position failed");
}
```

line 313 ~319

```c++
void AudioPlayer::HandleResult(const std::string &func_name, int result) {
  if (result != PLAYER_ERROR_NONE) {
    std::string error = ErrorToString(result);
    LOG_ERROR("%s failed : %s", func_name.c_str(), error.c_str());
    throw AudioPlayerError(error, func_name + " failed");
  }
}
```

line 378 ~382

```c++
void AudioPlayer::OnErrorOccurred(int code, void *data) {
  std::string error = ErrorToString(code);
  LOG_ERROR("error occurred: %s", error.c_str());
  AudioPlayer *player = (AudioPlayer *)data;
  player->error_listener_(player->player_id_, "error occurred: " + error);
}
```

For me, having a helper function looks better than creating a string type of `get_error_message` multiple times.
Of course, there might be a better way, and I'm open to other suggestions.

Reply:
Hmm. I think

```cpp
if (result != PLAYER_ERROR_NONE) {
  seeking_ = false;
  LOG_ERROR("player_set_play_position failed : %s", get_error_message(result));
  throw AudioPlayerError(get_error_message(result), "player_set_play_position failed");
}
```

or

```cpp
void AudioPlayer::OnErrorOccurred(int code, void *data) {
  std::string error(get_error_message(code));
  LOG_ERROR("error occurred: %s", error.c_str());
  AudioPlayer *player = (AudioPlayer *)data;
  player->error_listener_(player->player_id_, "error occurred: " + error);
}
```

should just work. There's no problem with the current style though. I just wanted to show another option in case you're going to apply this to all other plugins.

Reply:
Actually those look good, I will apply similar fixes to other plugins. Thanks!